### PR TITLE
 BugFix: when clicked on one of the items of the sort dropdown, it is…

### PR DESCRIPTION
**Pull request by salimshamim**
When clicking on an item, the select value changes so quickly that the popper disappears and the touch event simultaneously clicks the element behind it

### PR Fixes:
- 1 Issue 459

Resolves #[[459](https://github.com/code100x/daily-code/issues/459)]

**To solve the issue with minimal code change, introduced a delay of 20(twenty) milliseconds. This doesn't seem to hamper UX also fixes the issue**

### Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] I assure there is no similar/duplicate pull request regarding same issue

Solution is hacky but minimally.
Links to a [broader issue](https://github.com/radix-ui/primitives/issues/1658#issuecomment-1517705980)

### CODE REVIEW

**Comment by devsargam** [comment]
LGTM, testing it locally

**Comment by devsargam** [comment]
Hey @salimshamim can u fix the conflicts???

**Comment by devsargam** [comment]
Gentle remainder @salimshamim if you could fix conflicts

**Comment by salimshamim** [comment]
@devsargam Sorry, it was long time since I PRd it. Conflicts have been resolved.

File: apps/web/components/Tracks.tsx [hunk 1]
```diff
@@ -21,6 +21,8 @@ export const Tracks = ({ tracks }: { tracks: TrackPros[] }) => {
   const selectedCategory = useRecoilValue(category);
   const [filteredTracks, setFilteredTracks] = useState<TrackPros[]>(tracks);
   const [sortBy, setSortBy] = useState<string>("");
+
```
Comment (devsargam) [inline apps/web/components/Tracks.tsx:24]:
```suggestion
```

File: apps/web/components/Tracks.tsx [hunk 2]
```diff
@@ -21,6 +21,8 @@ export const Tracks = ({ tracks }: { tracks: TrackPros[] }) => {
...
+  const [selectOpen, setSelectOpen] = useState<boolean>(false);
```
Comment (devsargam) [inline apps/web/components/Tracks.tsx:25]:
```suggestion
  const [isSelectOpen, setIsSelectOpen] = useState<boolean>(false);
```

File: apps/web/components/Tracks.tsx [hunk 3]
```diff
@@ -56,6 +58,8 @@ export const Tracks = ({ tracks }: { tracks: TrackPros[] }) => {
     <div>
       <div className="flex flex-col gap-4 md:flex-row items-center justify-evenly mt-6">
         <Select
+          open = {selectOpen}
```
Comment (devsargam) [inline apps/web/components/Tracks.tsx:61]:
```suggestion
          open = {isSelectOpen}
```

File: apps/web/components/Tracks.tsx [hunk 4]
```diff
@@ -56,6 +58,8 @@ export const Tracks = ({ tracks }: { tracks: TrackPros[] }) => {
...
+          onOpenChange={() => { setTimeout(() => { setSelectOpen(!selectOpen); }, 20); }}
```
Comment (devsargam) [inline apps/web/components/Tracks.tsx:62]:
```suggestion
          onOpenChange={() => { setTimeout(() => { setIsSelectOpen(!isSelectOpen); }, 20); }}
```

**Review by devsargam — APPROVED** [review]
Few nits. Looks good

**Review by devsargam — APPROVED** [review]

**Comment by devsargam** [comment]
Thank you @salimshamim for your contribution. I don't think if it deserves a bounty since it's a fairly small change

**Pull request merged**